### ISSUE TITLE
Re-activate using Maning rather than Strickler?

### ISSUE THREAD

**Issue by benRenard** [BaRatin-tools/BaRatinAGE#42]
The  "Manning" option was deactivated in v2.2 because it wasn't implemented properly 

Implementing it properly is tricky: it would require rethinking the implementation of several key objects and changing the way studies are saved. Is it worth the candle?

Note that it's still possible to use the 'Manning prior assistant', but it's just an intermediate to compute a prior on Ks.

**Comment by IvanHeriver** [comment]
I think we should keep this for a future feature of BaRatinAGE v3. 

It should therefore be relabeled as an "enhancement" and renamed to "add the possibility to use Manning instead of Strickler for roughness" or something similar.

**Comment by IvanHeriver** [comment]
Just for reference, this issue is somewhat related to issue #25

**Comment by JeromeLeCoz** [comment]
I'm not sure I understand why you need to transform the prior distribution of the Manning's n... In the prior assistant, we only apply a propagation of variance to compute the prior uncertainty of coefficient "a". Having K_s or 1/n in the equation doesn't change anything imo, you just sum u²_n instead of u²_Ks... or am I wrong?

**Comment by JeromeLeCoz** [comment]
I played with the R script though, and did not find much difference between the distrib and its approximation...

**Comment by IvanHeriver** [comment]
> Yes the formula is correct, although if I understand correctly you want to do the opposite (i.e. the user provides a Manning prior and you want to translate it as a Strickler prior that will be used under the hood):

 actually I do it both ways: when the user specify the Manning coefficient, the Strickler coefficient is set this way, if the user specifies the Strickler Coefficient, then the Manning coefficient is set this way. They are always kept in sync. 

But in the end, only the Strickler coefficient is used in the computation of `a` to have consistent results regardless of whether the user is using Manning or Strickler i.e. changing the Manning/Strickler specification option should not change the result for `a`. But maybe I'm wrong and we should have different computation for the uncertainty of `a` since the propgation formula are different (see the piece of java code below) 

> I played with the R script though, and did not find much difference between the distrib and its approximation...

same here. 

@JeromeLeCoz : 

- The user may change from Manning to Strickler (and the otherway around) at any time so I need a way to set the uncertainty of one using the uncertainty of the other. Does that make sens?
- computing the uncertainty of `a` is different if using Manning or Strickler (see below the Java code I came up with); don't know if that's relevant to our question...

```java

        double Astd_Strickler = Math.sqrt(
                Math.pow(W * sqrtOfSlope * Kstd, 2) +
                Math.pow(K * sqrtOfSlope * Wstd, 2) +
                Math.pow(K * W / (2 * sqrtOfSlope) * Sstd, 2)
       );

        double Astd_Manning = Math.sqrt(
                Math.pow(W * sqrtOfSlope / (N * N) * Nstd, 2) +
                Math.pow(sqrtOfSlope / N * Wstd, 2) +
                Math.pow(W / (2 * N * sqrtOfSlope) * Sstd, 2)
        );

```

**Comment by IvanHeriver** [comment]
Mmmh, I see. Sadly, this is not what I implemented. And from what I understand, the Manning/Strickler choice should be made for each channel control (with the default being set in the preference) and changing it should unset the parameter (avoiding all what we are discussing here). Then `a` should be computed differently.

Currently changing the Manning/Strickler choice in the preferences changes all the existing rougness fields.

**Comment by JeromeLeCoz** [comment]
Yes, but if simpler to implement, I don't see it as a problem that the Manning/Strickler choice be fixed for a given study. I can't imagine a mad user shifting from Manning to Strickler throughout the study, across controls.
